nomad/acl_endpoint.go

```diff
@@ -423,7 +388,7 @@ func (a *ACL) GetPolicies(args *structs.ACLPolicySetRequest, reply *structs.ACLP
 					continue
 				}
 
-				if token.Type != structs.ACLManagementToken && !tokenPolicyNames.Contains(policyName) {
+				if !mgt && !tokenPolicyNames.Contains(policyName) {
```

Comment:
the loosely-coupled relationship between the `mgt` `bool` and the `nil`-ness of `tokenPolicyNames` has me nervous about accidentally causing a panic...  I would like to think a test would blow up if there was a code error like that, but I'm still like 😬

up in `ListPolicies` it's `if mgt || tokenPolicyNames.Contains(...`

`GetPolicies` doesn't trigger me, because they are directly coupled -- `tokenPolicyNames` is only created, and used, within an `if !mgt{}`

sometimes I worry I'm over-sensitive about nil panics

Reply:
> the code is obviously correct

I'm with you there. my concern with these things is usually about how easily a future person might accidentally cause it to break. if that case is covered by tests (here I suspect it is), then my concern is a lot less, because the problem is easily discovered.

Reply:
Implemented in 694785c181c01cecb51565f6653faa9bedbafd88. This further encapsulates the management vs non-management distinction and returns a function that can be used to do the check. (It avoids allocating the set if we don't need it, but we have the function pointer so that's mostly a wash.)